src/ff1.rs

```diff
@@ -268,11 +264,10 @@ impl<CIPH: BlockCipher + BlockEncrypt + Clone> FF1<CIPH> {
         let t = tweak.len();
 
         // 1. Let u = floor(n / 2); v = n - u
-        let u = n / 2;
-        let v = n - u;
-
         // 2. Let A = X[1..u]; B = X[u + 1..n].
-        let (mut x_a, mut x_b) = x.split(u);
+        let (mut x_a, mut x_b) = x.split();
+        let u = x_a.numeral_count();
+        let v = x_b.numeral_count();
```

Comment:
I don't understand how this works given that `BinaryNumeralString::numeral_count()` always returns a multiple of 8, but $u$ and $v$ might be multiples of 4 but not 8 (such as for the use in Zcash where they are 44).

Reply:
`x_a` and `x_b` are not of type `BinaryNumeralString`. That's one of the changes I make in this PR. I think their type is now `BinaryOps` (but I'm AFK and can't check).